### PR TITLE
Adds Component prop to ConnectedComponents in Onboarding

### LINKED ISSUE #774: Update ConnectedComponents to accept a component prop
### User Story

As an engineer I want to be able to swap out a ConnectedComponents rendered "child" so that I can reuse ConnectedComponents in the same way we can regular Components.

### Why
Our coding architecture encourages simple modular pure components that can be reused that are "driven by a ConnectedComponent. Currently connected components are hardcoded and tied to render their pure "child" component that they fetch data for. This makes it impossible to reuse them or customize the rendered child while keeping the ConnectedComponent unchanged resulting in unnecessary copy/paste and code debt. 

### Acceptance Criteria
Decouple ConnectedComponents from their rendered children.

### Possible Implementation & Open Questions
- add a `component` prop to ConnectedComponents
- default `component` to the current rendered "child"

### PR DESCRIPTION
**Pull request by burkeshartsis**
### What does this PR do, or why is it needed?
This PR adds a `Component` prop to all of the onboarding ConnectedComponents.

### What design trade-offs/decisions were made?
None. All of the ConnectedComponents just default to the components they previously were rendering as children.

### How do I test this PR?
Open app launch onboarding. Everything works? Your good!

### What automated tests did you write?
The new `Component` prop has 100% snapshot coverage 💥

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Partial work towards #774 
- [x] No new warnings in tests, in storybook, and in-app
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-ui-onboarding/src/slides/AboutYou/AboutYouConnected.js [hunk 1]
```diff
@@ -94,10 +94,20 @@ const AboutYouConnected = memo(
...
+    PropTypes.node,
+    PropTypes.func,
+    PropTypes.object,
+  ]),
   onPressPrimary: PropTypes.func,
   onPressSecondary: PropTypes.func,
 };
 
+AboutYouConnected.defaultProps = {
+  Component: AboutYou,
```
Comment (vinnyjth) [inline packages/apollos-ui-onboarding/src/slides/AboutYou/AboutYouConnected.js]: I disagree with default props being here, like we talked about. I think it leads to confusion with no additional benefit, for functional components, I would prefer if we start setting defaults via `propName = default` in the function arguments.

File: packages/apollos-ui-onboarding/src/slides/AboutYou/AboutYouConnected.tests.js [hunk 2]
```diff
@@ -44,4 +45,38 @@ describe('AboutYouConnected component', () => {
     );
     expect(tree).toMatchSnapshot();
   });
+  it('should render a custom Component', async () => {
```
Comment (vinnyjth) [inline packages/apollos-ui-onboarding/src/slides/AboutYou/AboutYouConnected.tests.js]: Nice 👍

File: packages/apollos-ui-onboarding/src/slides/AskNotifications/AskNotificationsConnected.tests.js [hunk 3]
```diff
@@ -47,7 +49,35 @@ describe('The Onboarding AskNotificationsConnected component', () => {
...
+          data: { notificationsEnabled: true },
+        },
+      },
+    ];
 
+    const CustomComponent = (
+      { buttonDisabled } // eslint-disable-line react/prop-types
+    ) => (
+      // `notificationsEnabled` is passed to the `buttonDisabled` prop thus it is equal to `true` from the mock query above.
+      <Text>{`Skyline is better than Goldstar === ${buttonDisabled}`}</Text>
```
Comment (vinnyjth) [inline packages/apollos-ui-onboarding/src/slides/AskNotifications/AskNotificationsConnected.tests.js]: lol

File: packages/apollos-ui-onboarding/src/slides/Features/FeaturesConnected.tests.js [hunk 4]
```diff
@@ -1,34 +1,78 @@
...
+              id: 'Person:123',
               firstName: 'Marty',
             },
           },
         },
       },
     };
-    const tree = renderer.create(
+
+    const tree = await renderWithApolloData(
```
Comment (vinnyjth) [inline packages/apollos-ui-onboarding/src/slides/Features/FeaturesConnected.tests.js]: Thanks for removing `waait` here

File: packages/apollos-ui-onboarding/src/slides/AboutYou/AboutYouConnected.js (hunk 1 above)
Comment (burkeshartsis) [reply packages/apollos-ui-onboarding/src/slides/AboutYou/AboutYouConnected.js]: And yet you merged the PR anyways! 😘 

@conrad-vanl what do you think of this moving forward?

**Pull request merged**